Mathlib/Data/Complex/Basic.lean

```diff
@@ -757,11 +757,60 @@ protected theorem mul_inv_cancel {z : ℂ} (h : z ≠ 0) : z * z⁻¹ = 1 := by
     ofReal_one]
 #align complex.mul_inv_cancel Complex.mul_inv_cancel
 
-/-! ### Field instance and lemmas -/
+noncomputable instance : RatCast ℂ where
+  ratCast := Rat.castRec
+
+/-! ### Cast lemmas -/
+
+@[simp, norm_cast]
+theorem ofReal_nat_cast (n : ℕ) : ((n : ℝ) : ℂ) = n :=
+  map_natCast ofReal n
+#align complex.of_real_nat_cast Complex.ofReal_nat_cast
 
+@[simp, norm_cast]
+theorem nat_cast_re (n : ℕ) : (n : ℂ).re = n := by rw [← ofReal_nat_cast, ofReal_re]
+#align complex.nat_cast_re Complex.nat_cast_re
+
+@[simp, norm_cast]
+theorem nat_cast_im (n : ℕ) : (n : ℂ).im = 0 := by rw [← ofReal_nat_cast, ofReal_im]
+#align complex.nat_cast_im Complex.nat_cast_im
+
+@[simp, norm_cast]
+theorem ofReal_int_cast (n : ℤ) : ((n : ℝ) : ℂ) = n :=
+  map_intCast ofReal n
+#align complex.of_real_int_cast Complex.ofReal_int_cast
+
+@[simp, norm_cast]
+theorem int_cast_re (n : ℤ) : (n : ℂ).re = n := by rw [← ofReal_int_cast, ofReal_re]
+#align complex.int_cast_re Complex.int_cast_re
+
+@[simp, norm_cast]
+theorem int_cast_im (n : ℤ) : (n : ℂ).im = 0 := by rw [← ofReal_int_cast, ofReal_im]
+#align complex.int_cast_im Complex.int_cast_im
+
+@[simp, norm_cast]
+theorem rat_cast_im (q : ℚ) : (q : ℂ).im = 0 := by
+  show (Rat.castRec q : ℂ).im = 0
+  cases q
+  simp [Rat.castRec]
+#align complex.rat_cast_im Complex.rat_cast_im
+
+@[simp, norm_cast]
+theorem rat_cast_re (q : ℚ) : (q : ℂ).re = (q : ℝ) := by
+  show (Rat.castRec q : ℂ).re = _
+  cases q
+  simp [Rat.castRec, normSq, Rat.mk_eq_divInt, Rat.mkRat_eq_div, div_eq_mul_inv, *]
+#align complex.rat_cast_re Complex.rat_cast_re
+
+/-! ### Field instance and lemmas -/
 
 noncomputable instance : Field ℂ :=
-{ inv := Inv.inv
+{
+/- We define `qsmul` in this way to ensure it is equal to the more general instance
+`SMul R ℝ → SMul R ℂ` found in `Data.Complex.Module`. -/
+  qsmul := fun n z => ⟨n • z.re - 0 * z.im, n • z.im + 0 * z.re⟩
```

Comment:
Why not just use
```suggestion
/- We define `qsmul` in this way to ensure it is equal to the more general instance
`SMul R ℝ → SMul R ℂ` found in `Data.Complex.Module`. -/
  qsmul := fun n z => n • z
```
if that instance already exists?

Reply:
The generic `SMul` instance is defined in `Data.Complex.Module` which is not imported

Reply:
Can we move the smul instance (and only that insurance) to this file?